### PR TITLE
[12.0] Backport tech_name mixin inheritance

### PR DESCRIPTION
**Pull request by len-foss**
- [ ] depends on https://github.com/OCA/server-env/pull/71

Before these commits, server env usage is kinda broken: es_server_host is not a server env field (despite the declaration), moreover the env key is the name, whereas se.backend uses the tech_name.

### CODE REVIEW

**Comment by simahawk** [comment]
@len-foss could you check travis?

**Comment by len-foss** [comment]
@simahawk PR https://github.com/OCA/server-env/pull/71 is required for this one to work, so there's no fixing Travis.

**Comment by simahawk** [comment]
@len-foss next time please mention it in the description ;) (see how I updated it)

**Comment by hparfr** [comment]
@PierrickBrun please have a look

**Comment by PierrickBrun** [comment]
@len-foss since https://github.com/OCA/server-env/pull/71 is merged, can you rerun travis/runbot ? With a rebase for instance

File: connector_search_engine/models/se_index.py [hunk 1]
```diff
@@ -117,13 +115,16 @@ def generate_batch_export_per_index(self, domain=None):
             record._jobify_batch_export()
         return True
 
-    def _get_domain_for_exporting_binding(self):
-        return [("index_id", "=", self.id), ("sync_state", "=", "to_update")]
+    def _get_domain_for_exporting_binding(self, force_export=False):
+        domain = [("index_id", "=", self.id)]
+        if not force_export:
+            domain.append(("sync_state", "=", "to_update"))
```
Comment (PierrickBrun) [inline connector_search_engine/models/se_index.py:121]: I think this breaks a test

**Review by PierrickBrun — APPROVED** [review]
functional test

**Comment by len-foss** [comment]
@PierrickBrun I'm sorry, I still didn't have the time to come back to this, and I'm still a bit too far from Shopinvader and that kind of stuff for now. Hopefully that should happen in the coming weeks.

File: connector_search_engine/models/se_index.py [hunk 2]
```diff
@@ -93,21 +93,19 @@ def _compute_name(self):
...
-        bindings.write({"sync_state": "to_update"})
-        self._jobify_batch_export()
+        self._jobify_batch_export(force_export=True)
 
-    def _jobify_batch_export(self):
+    def _jobify_batch_export(self, force_export=False):
         self.ensure_one()
         description = _("Prepare a batch export of index '%s'") % self.name
-        self.with_delay(description=description).batch_export()
+        self.with_delay(description=description).batch_export(force_export)
```
Comment (PierrickBrun) [inline connector_search_engine/models/se_index.py:108]:
```suggestion
        self.with_delay(description=description).batch_export(force_export=force_export)
```

File: connector_search_engine/models/se_index.py [hunk 3]
```diff
@@ -117,13 +115,16 @@ def generate_batch_export_per_index(self, domain=None):
...
+        if not force_export:
+            domain.append(("sync_state", "=", "to_update"))
+        return domain
 
     @job(default_channel="root.search_engine.prepare_batch_export")
-    def batch_export(self):
+    def batch_export(self, force_export=False):
         self.ensure_one()
-        domain = self._get_domain_for_exporting_binding()
+        domain = self._get_domain_for_exporting_binding(force_export)
```
Comment (PierrickBrun) [inline connector_search_engine/models/se_index.py:127]:
```suggestion
        domain = self._get_domain_for_exporting_binding(force_export=force_export)
```

**Comment by PierrickBrun** [comment]
> @PierrickBrun I'm sorry, I still didn't have the time to come back to this, and I'm still a bit too far from Shopinvader and that kind of stuff for now. Hopefully that should happen in the coming weeks.

No problem, I did a functional test and it works, also I think I found what broke the test

**Pull request closed without merging**